src/prpy/perception/block_detector.py

```diff
@@ -0,0 +1,201 @@
+        @param min_cluster_size: minimum number of points in a cluster
+        @param max_cluster_size: maximum number of points in a cluster
+        @param segment_box: flag to discard points outside of (box_min, box_max)
+        @param box_min: minimum coordinsates of search area in camera frame
+        @param box_max: maximum coordinsates of search area in camera frame
+        """
+
+        print "waiting for service..."
+        rospy.wait_for_service(service_name);
+        print "Calling service..."
+        try:
+            box_min_pt = geometry_msgs.msg.Point();
+            box_min_pt.x = box_min[0];
+            box_min_pt.y = box_min[1];
+            box_min_pt.z = box_min[2];
+            box_max_pt = geometry_msgs.msg.Point();
+            box_max_pt.x = box_max[0];
+            box_max_pt.y = box_max[1];
+            box_max_pt.z = box_max[2];
+
+            service = rospy.ServiceProxy(service_name, FindBlocks);
+            response = service(cloud_topic, segment_planes, num_planes, plane_distance, segment_depth, min_depth, max_depth, cluster_tolerance, min_cluster_size, max_cluster_size, segment_box, box_min_pt, box_max_pt);
+            return response.blocks;
+        except rospy.ServiceException, e:
+            print "Service call failed: %s"%e
+            return []
+
+
+    @PerceptionMethod
+    def DetectObjects(self, robot, **kw_args):
+        """
+        Detect the table and the bin
+        """
+        env = robot.GetEnv()
+
+        # Get the pr-ordata package path
+        data_dir = prpy.util.FindCatkinResource('pr_ordata', 'data')
+
+        # Place the table in the environment
+        #table_path = os.path.join(data_dir, 'furniture', 'table.kinbody.xml')
+        #table = env.ReadKinBodyXMLFile(table_path)
+        #env.Add(table)
+        
+        #table_in_robot = numpy.array([[0., 0., 1., 1.025],
+        #                              [1., 0., 0., 0.],
+        #                              [0., 1., 0., 0.],
+        #                              [0., 0., 0., 1.]])
+        #table_in_world = numpy.dot(robot.GetTransform(), table_in_robot)
+        #table.SetTransform(table_in_world)
+        #table_aabb = table.ComputeAABB()
+        #table_height = table_aabb.pos()[2] + table_aabb.extents()[2] 
+
+        # TODO: Add a bin to the edge of the table for placing the blocks into
+        #tray_path = os.path.join(data_dir, 'objects', 'wicker_tray.kinbody.xml')
+        #tray = env.ReadKinBodyXMLFile(tray_path)
+        #tray_aabb = tray.ComputeAABB()
+        #env.Add(tray)
+
+        #xpose = table
+        #tray_in_table = numpy.array([[1., 0., 0., -0.7607], # right edge - -.99575 + .235
+        #                             [0., 0., 1., table_height + 0.01],
+        #                             [0.,-1., 0., 0.],
+        #                             [0., 0., 0., 1.]])
+        #tray_in_world = numpy.dot(table.GetTransform(), tray_in_table)
+        #tray.SetTransform(tray_in_world)
```

Comment:
I don't think this code is necessary now that we perceive the table. Can we delete it?